packages/kbn-esql-validation-autocomplete/src/validation/__tests__/fields_and_variables.test.ts

```diff
@@ -0,0 +1,218 @@
+      },
+      { name: 'DROP', command: 'FROM index | DROP kubernetes.`something`.`something`' },
+      {
+        name: 'ENRICH',
+        command:
+          'FROM index | ENRICH policy WITH `new`.name1 = `otherField`, `new.name2` = `yetAnotherField`',
+      },
+      { name: 'EVAL', command: 'FROM index | EVAL kubernetes.`something`.`something` + 12' },
+      {
+        name: 'GROK',
+        command: 'ROW `funky`.`stri#$ng` = "lolz" | GROK `funky`.`stri#$ng` "%{WORD:firstWord}"',
+      },
+      { name: 'KEEP', command: 'FROM index | KEEP kubernetes.`something`.`something`' },
+      {
+        name: 'RENAME',
+        command: 'FROM index | RENAME kubernetes.`something`.`something` as foobar',
+      },
+      { name: 'SORT', command: 'FROM index | SORT kubernetes.`something`.`something` DESC' },
+      {
+        name: 'STATS ... BY',
+        command:
+          'FROM index | STATS AVG(kubernetes.`something`.`something`) BY `kubernetes`.`something`.`something`',
+      },
+      { name: 'WHERE', command: 'FROM index | WHERE kubernetes.`something`.`something` == 12' },
+    ];
+
+    it.each(cases)('$name accepts escaped fields', async ({ command }) => {
+      const { expectErrors } = await setup();
+      await expectErrors(command, []);
+    });
+  });
+});
+
+describe('variable support', () => {
+  describe('variable data type detection', () => {
+    // most of these tests are aspirational (and skipped) because we don't have
+    // a good way to compute the type of an expression yet.
+    beforeAll(() => {
+      setTestFunctions([
+        // this test function is just used to test the type of the variable
+        {
+          type: 'eval',
+          description: 'Test function',
+          supportedCommands: ['eval'],
+          name: 'test',
+          signatures: [
+            { params: [{ name: 'arg', type: 'cartesian_point' }], returnType: 'cartesian_point' },
+          ],
+        },
+        // this test function is used to check that the correct return type is used
+        // when determining variable types
+        {
+          type: 'eval',
+          description: 'Test function',
+          supportedCommands: ['eval'],
+          name: 'return_value',
+          signatures: [
+            { params: [{ name: 'arg', type: 'text' }], returnType: 'text' },
+            { params: [{ name: 'arg', type: 'double' }], returnType: 'double' },
+            {
+              params: [
+                { name: 'arg', type: 'double' },
+                { name: 'arg', type: 'text' },
+              ],
+              returnType: 'long',
+            },
+          ],
+        },
+      ]);
+    });
+
+    afterAll(() => {
+      setTestFunctions([]);
+    });
+
+    const expectType = (type: FunctionParameterType) =>
+      `Argument of [test] must be [cartesian_point], found value [var] type [${type}]`;
+
+    test.skip('literals', async () => {
```

Comment:
Are these tests meant to be skipped?

Reply:
Ok I see the comment now in the describe block. Suggestion to add a todo note for each of the skipped tests or potentially refactor to skip the entire block. (Looks like the only test in this block we are testing is `fields`)

Reply:
Yeah, we can't support these cases yet. We will be able to after https://github.com/elastic/kibana/issues/195682

Reply:
Added some extra comments